src/runtime/test_case_runner.js

```diff
@@ -192,6 +224,28 @@ export default class TestCaseRunner {
     return this.invokeStep(null, hookDefinition, hookParameter)
   }
 
+  async runStepHooks(step, isBeforeHook) {
+    let stepHooks = []
+    const stepInfo = {
+      text: step.text,
+      arguments: step.arguments,
+      location: Object.assign({ uri: this.testCase.uri }, step.locations[0]),
+    }
```

Comment:
I'd suggest for beforeStep the hookParameter is `{step, testCase: {sourceLocation, pickle}}` where source location / pickle is what we pass to before / after hooks. And for afterStep we use `{step, testCase: {sourceLocation, pickle}, result}` where result is the step result.

Reply:
Ok, I will have a look towards the end of this week.

Reply:
I think I found a solution. Still need to test it, but will update the pr soon.

Reply:
@charlierudolph, updated the branch. It's not complete, I didn't add additional tests, just made sure nothing breaks. Can you please check it and let me know if you're ok with the solution?

Reply:
Sorry way late here. The current solution looks good to me. I like the cumulation of before step / step / after step results together. May be worth creating a "test_step_runner" class to encompass the new logic and to make testing a bit easier.

Also note, a lot of things have changed since you last merged master. (Gherkin 9 + cucumber-messages and the conversion to typescript). If you'd like help migrating in the latest changes, let me know.